Assets/MRTK/SDK/Features/Utilities/Solvers/TapToPlace.cs

```diff
@@ -75,7 +75,42 @@ public float MaxRaycastDistance
         public float SurfaceNormalOffset
         {
             get => surfaceNormalOffset;
-            set => surfaceNormalOffset = value;
+            set
+            {
+                if (surfaceNormalOffset != value)
+                {
+                    // If the current surface normal offset is the defaultSurfaceNormalOffset set the UseDefaultSurfaceNormalOffset
+                    if (value == defaultSurfaceNormalOffset)
```

Comment:
I would change this setter to be a 'dumb' setter (i.e. it just sets the surfaceNormalOffset value and doesn't try to compare it against defaultSurfaceNormalOffset).

Having a property that implicitly changes other properties based on alignment with magic values (in this case defaultSurfaceNormalOffset) violates the principle of least surprise. As a caller of this, I wouldn't expect that changing a float value of something would also change the bool value of something else - i.e. as the caller of any thing that modifies a property, I would expect that to only modify that property.

Otherwise, consider a case where a caller is maybe cycling through changing the surface normal offset and they HAPPEN to set it to defaultSurfaceNormalOffset on a particular frame - well, now the next frame where they want to set it to offset + .001, it stops working. Why? Because UseDefaultSurfaceNormalOffset was just set to true and they didn't know it happened.

So now every single consumer has

Reply:
@wiwei I understand the least surprise principle concern, but I think in this case it is necessary to connect `UseDefaultSurfaceNormalOffset` and `SurfaceNormalOffset` to minimize the probability of confusion.

Scenario 1:
Someone to sets the `SurfaceNormalOffset` to the exact value as default, which will set `UseDefaultSurfaceNormalOffset` to true without intention from the developer.
  - In the scenario you described where a user is cycling through values for `SurfaceNormalOffset`, if they happen to land of the defalutSurfaceNormalOffset, `UseDefaultSurfaceNormalOffset` will be set to true, but this does not affect setting other values.  They will see that `UseDefaultSurfaceNormalOffset` is true for that frame, but then once they change the `SurfaceNormalOffset` again, `UseDefaultSurfaceNormalOffset` will go back to false.  Yes it would be confusing to see, but not as likely as scenario 2.

Scenario 2:
Someone sets the `SurfaceNormalOffset` while `UseDefaultSurfaceNormalOffset` is still true.  
  - This is very likely because by default `UseDefaultSurfaceNormalOffset` is true.  I can see someone wondering why their new value for `SurfaceNormalOffset` is not getting applied, but what is happening is `UseDefaultSurfaceNormalOffset` is still true which overrides their new `SurfaceNormalOffset` value.

Reply:
I feel pretty strongly about disconnecting these two. The current way also makes it impossible to set the SurfaceNormalOffset without turning off UseDefaultSurfaceNormalOffset (i.e. you want to have one part of the code that changes the offset amount, and have something else that applies policy for when that value should actually be used).

Reply:
I'm inclined to agree with @wiwei here. I think the main confusion here stems from the fact that a public **Surface Normal Offset** property is actually getting updated in real-time whenever **Use Default Surface Normal Offset** is true. I actually wouldn't expect this. I would expect the **Surface Normal Offset** property to be whatever I set it to at all times. If I never set it, I would expect it to be 0.0f because that is the default value for float.

Conceptually I think the **Surface Normal Offset** property only has any meaning if **Use Default Surface Normal Offset** is false. And this mental model is supported by the inspector hiding that field when the default is in use. Therefore I don't think the default calculation should update the public property at all. I think that should just happen behind the scenes and the two should be completely unlinked from each other.

Reply:
Ok, I will separate the properties.  The original intention was to give control to `UseDefaultSurfaceNormalOffset` to determine whether or not `SurfaceNormalOffset` could be set, but now I see how this could cause confusion.